### ISSUE TITLE
[Bug]: CLI not uploading files upon save

### ISSUE THREAD

**Issue by lspoor** [Shopify/shopify-cli#2457]
### Please confirm that you:

- [X] Are working with Shopify CLI 2.0, not 3.0. (You can [open a 3.0 issue here](https://github.com/Shopify/cli/issues/new/choose).)
- [X] Searched [existing issues](https://github.com/Shopify/shopify-cli/issues) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://github.com/Shopify/shopify-cli/releases).

### What type of project are you working on?

theme

### Expected behavior

Upon saving a file, the change would be uploaded to the development theme (while in `shopify theme serve`)

### Actual behavior

Nothing actually happens, no upload of the change happens.

### Reproduction steps

1. Run `shopify theme serve`
2. Make a change in a file and save
3. Check the terminal to see if it shows if uploaded, as well as the theme within preview to see if the change it there

### Operating System

macOS Montery (M2 Chip)

### CLI version (run `shopify version` if you're not sure)

2.20.0

### Shell

fish

### Ruby version (run `ruby -v` if you're not sure)

3.0.0

### What language and version are you using in your application?

Node v18.5.0

**Comment by lspoor** [comment]
@jessetuglu Was your latest pull request supposed to fix this? I've just upgraded to `2.20.1` and still encountering the issue, no files being uploading on save.

**Comment by lspoor** [comment]
Looks like it is watching the correct folder

```
I, [2022-07-19T12:57:35.306156 #5312]  INFO -- : Record.build(): 0.01114 seconds
I, [2022-07-19T12:57:35.309682 #5312]  INFO -- : fsevent: watching: ["/Users/lukespoor/Sites/shopify-webpack"]
```
<img width="1550" alt="Screenshot 2022-07-19 at 12 57 38" src="https://user-images.githubusercontent.com/1685176/179744696-aac490cf-cb30-4876-9519-be3c232efb99.png">

**Comment by lspoor** [comment]
@jessetuglu Any progress on this, updated to `2.21.0` and still no luck. Unable to do any work with this issue currently.

**Comment by lspoor** [comment]
Bit more coming through this time!

Hope this helps

<img width="1550" alt="Screenshot 2022-08-04 at 13 05 44" src="https://user-images.githubusercontent.com/1685176/182842836-b955b580-3229-4188-bf09-944a2d3efa3b.png">

**Comment by jessetuglu** [comment]
This is great, thank you @lspoor. Would you be able to send your zipped theme(including ignored files) to `jesse.tuglu@shopify.com`? I'm wondering if there's a file that's not being picked up/ignored by the zipped export from your store.

Regarding the CLI versioning:
1. `git clone https://github.com/Shopify/shopify-cli` into a directory of your choice.
2. Inside the repo, type `git checkout 411c3324d09c3f1cfb0f72e79821e441f5c782a2`, this is the commit hash for the `2.18.0` release.
3. Into your `~./zshrc` or `~./bashrc`, type `alias shopify-dev='~/{path-to-shopify-cli-repo}/shopify-cli/bin/shopify'`
4. After reloading your terminal window, you should be able to go into any theme directory and run `shopify-dev theme serve` and your changes should sync as normal.

**Comment by lspoor** [comment]
Hey @jessetuglu, to make life easier I've added you as a collaborator for the repo. Invite should be with you!

Will try getting `2.18.0` running! Thanks!

**Comment by lspoor** [comment]
@jessetuglu Should this work with ruby `3.0.0p0`? 

Encountering this issue:

<img width="1550" alt="Screenshot 2022-08-04 at 18 45 16" src="https://user-images.githubusercontent.com/1685176/182916566-9c266ab7-c3a5-41df-aa0f-ddf7df0440b2.png">

**Comment by jessetuglu** [comment]
Yes, this is the same error I get usually when I'm on a different ruby version using `zsh`, if possible I would try using ruby `v3.1.0`. In the theme you shared, the `theme.liquid` file is located at: `src/liquid/layout/theme.liquid` – is something you changed since the latest commit?

**Comment by lspoor** [comment]
So the `/src` files are there when I'm using webpack, so `npm run start` includes `shopify theme serve`. For now I'm just working directly out from `/` so `/layout/theme.liquid` while having this issue. But webpack would save copy the files from `/src` to `/` and then upload :)

**Comment by jessetuglu** [comment]
Got it, I'll get that working now 👍. You may have to head into the `shopify-cli` directory (the one which you cloned) and run `bundle install`(you may have to install bundler: `gem install bundler`). This should give you all the gems on `v3.1.0` to run the command again in the theme directory. Thanks for your patience! 🙏

**Comment by jessetuglu** [comment]
You bet! Let me know if you run into any more issues – happy to help.
